### PR TITLE
[CARBONDATA-1189] Fixed delete with subquery issue

### PR DESCRIPTION
**Pull request by ravipesala**
The reason is tupleid udf is not properly pushed down to the projection plan in case of join condition.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2563/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/456/<h2>Failed Tests: <span class='status-failure'>2</span></h2><h3><a name='carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test' /><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/456/org.apache.carbondata$carbondata-spark-common-test/testReport'>carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test</a>: <span class='status-failure'>2</span></h3><ul><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/456/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/DeleteCarbonTableTestCase/delete_data_from__carbon_table_where_IN__sub_query___/'><strong>org.apache.carbondata.spark.testsuite.iud.DeleteCarbonTableTestCase.delete data from  carbon table[where IN (sub query) ]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/456/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/DeleteCarbonTableTestCase/delete_data_from__carbon_table_where_IN__sub_query_with_where_clause___/'><strong>org.apache.carbondata.spark.testsuite.iud.DeleteCarbonTableTestCase.delete data from  carbon table[where IN (sub query with where clause) ]</strong></a></li></ul>

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2565/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/458/

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/DeleteCarbonTableTestCase.scala [hunk 1]
```diff
@@ -25,8 +25,6 @@ import org.apache.carbondata.core.util.CarbonProperties
 
 class DeleteCarbonTableTestCase extends QueryTest with BeforeAndAfterAll {
   override def beforeAll {
-    CarbonProperties.getInstance()
-      .addProperty(CarbonCommonConstants.ENABLE_VECTOR_READER , "false")
```
Comment (jackylk) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/DeleteCarbonTableTestCase.scala]: Why delete this, this bug occurs when using vector reader?

Comment (ravipesala) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/DeleteCarbonTableTestCase.scala]: It was disabled when vector reader in delete flow was not working. But now it is working so enabled it.  There is no relation with this bug with vector reader config.

**Comment by jackylk** [comment]
LGTM

**Pull request merged**